### PR TITLE
Switch to compiled handlebars

### PR DESCRIPTION
**Pull request by rullzer**
* Let webpack compile the templates
* Remove online compilation
* Recompile assets

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: src/webpack.common.js [hunk 1]
```diff
@@ -29,13 +29,21 @@ module.exports = {
...
+				}
 			}
 		]
 	},
 	plugins: [new VueLoaderPlugin()],
 	resolve: {
 		alias: {
-			vue$: 'vue/dist/vue.esm.js'
+			vue$: 'vue/dist/vue.esm.js',
+			'handlebars': 'handlebars/runtime.js'
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply src/webpack.common.js]: Works without, so I removed it again

**Comment by nickvergessen** [comment]
While this works for `make build-js-production` it doesn't work for `make build-js`.
The problem is that still uses `eval` in the compiled JS:
```
 [Mitteilungsdetails anzeigen/ausblenden] Error: call to eval() blocked by CSP notifications.js:882:1
./src/init.js
https://nextcloud15.local/appsbabies/notifications/js/notifications.js:882:1
__webpack_require__
https://nextcloud15.local/appsbabies/notifications/js/notifications.js:20:12
<anonym>
https://nextcloud15.local/appsbabies/notifications/js/notifications.js:84:18
<anonym>
https://nextcloud15.local/appsbabies/notifications/js/notifications.js:1:11
```

Code in question (882 is the eval line)
```js
/***/ "./src/init.js":
/*!*********************!*\
  !*** ./src/init.js ***!
  \*********************/
/*! no exports provided */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
eval("__webpack_require__.r(__webpack_exports__);\n/* harmony import */ var vue__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! vue */ \"./node_modules/vue/dist/vue.esm.js\");\n/* harmony import */ var _App__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ./App */ \"./src/App.vue\");\n/**\n * @copyright Copyright (c) 2018 Joas Schilling <coding@schilljs.com>\n *\n * @license GNU AGPL version 3 or any later version\n *\n * This program is free software: you can redistribute it and/or modify\n * it under the terms of the GNU Affero General Public License as\n * published by the Free Software Foundation, either version 3 of the\n * License, or (at your option) any later version.\n *\n * This program is distributed in the hope that it will be useful,\n * but WITHOUT ANY WARRANTY; without even the implied warranty of\n * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n * GNU Affero General Public License for more details.\n *\n * You should have received a copy of the GNU Affero General Public License\n * along with this program.  If not, see <http://www.gnu.org/licenses/>.\n *\n */\n\n\nvue__WEBPACK_IMPORTED_MODULE_0__[\"default\"].prototype.t = t;\nvue__WEBPACK_IMPORTED_MODULE_0__[\"default\"].prototype.n = n;\nvue__WEBPACK_IMPORTED_MODULE_0__[\"default\"].prototype.OC = OC;\nvue__WEBPACK_IMPORTED_MODULE_0__[\"default\"].prototype.OCA = OCA;\n$('form.searchbox').after($('<div>').attr('id', 'notifications'));\nnew vue__WEBPACK_IMPORTED_MODULE_0__[\"default\"]({\n  el: '#notifications',\n  render: function render(h) {\n    return h(_App__WEBPACK_IMPORTED_MODULE_1__[\"default\"]);\n  }\n});\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly8vLi9zcmMvaW5pdC5qcz9kOTllIl0sIm5hbWVzIjpbIlZ1ZSIsInByb3RvdHlwZSIsInQiLCJuIiwiT0MiLCJPQ0EiLCIkIiwiYWZ0ZXIiLCJhdHRyIiwiZWwiLCJyZW5kZXIiLCJoIiwiQXBwIl0sIm1hcHBpbmdzIjoiQUFBQTtBQUFBO0FBQUE7QUFBQTs7Ozs7Ozs7Ozs7Ozs7Ozs7OztBQW9CQTtBQUNBO0FBRUFBLDJDQUFHLENBQUNDLFNBQUosQ0FBY0MsQ0FBZCxHQUFrQkEsQ0FBbEI7QUFDQUYsMkNBQUcsQ0FBQ0MsU0FBSixDQUFjRSxDQUFkLEdBQWtCQSxDQUFsQjtBQUNBSCwyQ0FBRyxDQUFDQyxTQUFKLENBQWNHLEVBQWQsR0FBbUJBLEVBQW5CO0FBQ0FKLDJDQUFHLENBQUNDLFNBQUosQ0FBY0ksR0FBZCxHQUFvQkEsR0FBcEI7QUFFQUMsQ0FBQyxDQUFDLGdCQUFELENBQUQsQ0FBb0JDLEtBQXBCLENBQTBCRCxDQUFDLENBQUMsT0FBRCxDQUFELENBQVdFLElBQVgsQ0FBZ0IsSUFBaEIsRUFBc0IsZUFBdEIsQ0FBMUI7QUFFQSxJQUFJUiwyQ0FBSixDQUFRO0FBQ1BTLElBQUUsRUFBRSxnQkFERztBQUVQQyxRQUFNLEVBQUUsZ0JBQUFDLENBQUM7QUFBQSxXQUFJQSxDQUFDLENBQUNDLDRDQUFELENBQUw7QUFBQTtBQUZGLENBQVIiLCJmaWxlIjoiLi9zcmMvaW5pdC5qcy5qcyIsInNvdXJjZXNDb250ZW50IjpbIi8qKlxuICogQGNvcHlyaWdodCBDb3B5cmlnaHQgKGMpIDIwMTggSm9hcyBTY2hpbGxpbmcgPGNvZGluZ0BzY2hpbGxqcy5jb20+XG4gKlxuICogQGxpY2Vuc2UgR05VIEFHUEwgdmVyc2lvbiAzIG9yIGFueSBsYXRlciB2ZXJzaW9uXG4gKlxuICogVGhpcyBwcm9ncmFtIGlzIGZyZWUgc29mdHdhcmU6IHlvdSBjYW4gcmVkaXN0cmlidXRlIGl0IGFuZC9vciBtb2RpZnlcbiAqIGl0IHVuZGVyIHRoZSB0ZXJtcyBvZiB0aGUgR05VIEFmZmVybyBHZW5lcmFsIFB1YmxpYyBMaWNlbnNlIGFzXG4gKiBwdWJsaXNoZWQgYnkgdGhlIEZyZWUgU29mdHdhcmUgRm91bmRhdGlvbiwgZWl0aGVyIHZlcnNpb24gMyBvZiB0aGVcbiAqIExpY2Vuc2UsIG9yIChhdCB5b3VyIG9wdGlvbikgYW55IGxhdGVyIHZlcnNpb24uXG4gKlxuICogVGhpcyBwcm9ncmFtIGlzIGRpc3RyaWJ1dGVkIGluIHRoZSBob3BlIHRoYXQgaXQgd2lsbCBiZSB1c2VmdWwsXG4gKiBidXQgV0lUSE9VVCBBTlkgV0FSUkFOVFk7IHdpdGhvdXQgZXZlbiB0aGUgaW1wbGllZCB3YXJyYW50eSBvZlxuICogTUVSQ0hBTlRBQklMSVRZIG9yIEZJVE5FU1MgRk9SIEEgUEFSVElDVUxBUiBQVVJQT1NFLiAgU2VlIHRoZVxuICogR05VIEFmZmVybyBHZW5lcmFsIFB1YmxpYyBMaWNlbnNlIGZvciBtb3JlIGRldGFpbHMuXG4gKlxuICogWW91IHNob3VsZCBoYXZlIHJlY2VpdmVkIGEgY29weSBvZiB0aGUgR05VIEFmZmVybyBHZW5lcmFsIFB1YmxpYyBMaWNlbnNlXG4gKiBhbG9uZyB3aXRoIHRoaXMgcHJvZ3JhbS4gIElmIG5vdCwgc2VlIDxodHRwOi8vd3d3LmdudS5vcmcvbGljZW5zZXMvPi5cbiAqXG4gKi9cblxuaW1wb3J0IFZ1ZSBmcm9tICd2dWUnO1xuaW1wb3J0IEFwcCBmcm9tICcuL0FwcCc7XG5cblZ1ZS5wcm90b3R5cGUudCA9IHQ7XG5WdWUucHJvdG90eXBlLm4gPSBuO1xuVnVlLnByb3RvdHlwZS5PQyA9IE9DO1xuVnVlLnByb3RvdHlwZS5PQ0EgPSBPQ0E7XG5cbiQoJ2Zvcm0uc2VhcmNoYm94JykuYWZ0ZXIoJCgnPGRpdj4nKS5hdHRyKCdpZCcsICdub3RpZmljYXRpb25zJykpO1xuXG5uZXcgVnVlKHtcblx0ZWw6ICcjbm90aWZpY2F0aW9ucycsXG5cdHJlbmRlcjogaCA9PiBoKEFwcClcbn0pO1xuIl0sInNvdXJjZVJvb3QiOiIifQ==\n//# sourceURL=webpack-internal:///./src/init.js\n");

/***/ }),
```

Is this fixable @rullzer or out of your js/webpack knowledge?

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/notifications/blob/0fb58898baca4134af5d33b6314e2f25105733ec/src/webpack.dev.js#L11 -> pick a different one from https://webpack.js.org/configuration/devtool/

**Comment by nickvergessen** [comment]
Excellent, thanks

**Pull request merged**